packages/frontend/src/exploregens/App.vue

```diff
@@ -152,26 +155,27 @@ export default {
       return gen.state === "installed" ? "#585858" : "primary";
     },
     onAction(gen) {
-      if (!gen.disabledToHandle) {
-        const action = gen.state === "installed" ? "uninstall" : "install";
-        this.rpc.invoke(action, [gen]);
-      }
+      const action = gen.action.toLowerCase();
+      this.rpc.invoke(action, [gen]);
     },
-    onQueryChange() {
-      this.debouncedGenFilterChange();
+    onQueryChange: function () {
+      //TODO: use debounce this.debouncedGenFilterChange();
+      this.getFilteredGenerators();
```

Comment:
why ?

Reply:
It does not work with debounce
Need to investigate

Reply:
Fixed. Done